src/Swarm/Language/Pretty.hs

```diff
@@ -121,14 +121,27 @@ data Wildcard = Wildcard
 instance PrettyPrec Wildcard where
   prettyPrec _ _ = "_"
 
+-- | Split a function type chain, so that we can pretty print
+--   the type parameters aligned on each line when they don't fit.
+class UnchainableFun t where
+  unchainFun :: t -> [t]
+
+instance UnchainableFun Type where
+  unchainFun (a :->: ty) = a : unchainFun ty
+  unchainFun ty = [ty]
```

Comment:
@byorgey almost there, just have to figure out why it got too unchained 😅 
```
❯ cabal run swarm -O0 -- format --width 20 example.sw
def harvestbox:
  dir ->
  cmd unit ->
  cmd unit ->
  cmd unit ->
  cmd unit ->
  cmd unit 
  = \d. \rep1. \rep2.
  doboxP harvest d rep1 rep2 notempty
end
```

Reply:
Haha, oops!  That's odd.  The `UnchainableFun` instance above looks fine to me...

Hmm, maybe we should break up chains of function applications too...
```
def harvestbox
  = \d. \rep1. \rep2.
  doboxP 
    harvest 
    d 
    rep1
    rep2
    notempty
end
```

Reply:
@byorgey found it! I need to pass precedence to all but the last parameter.

Reply:
@byorgey I thought about function parameters too, but those might just be a bit complicated because of pattern matching operators. Let's leave that for a separate issue. 😅

Reply:
Oh, definitely leave it for a separate issue, I was just wondering out loud.

Reply:
I added it to the check list in #1571. 🙂